lib/web/scripts.go

```diff
@@ -0,0 +1,154 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package web
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/modules"
+	"github.com/gravitational/teleport/lib/utils/teleportassets"
+	"github.com/gravitational/teleport/lib/web/scripts"
+)
+
+// installScriptHandle handles calls for "/scripts/install.sh" and responds with a bash script installing Teleport
+// by downloading and running `teleport-update`. This installation script does not start the agent, join it,
+// or configure its services. This is handled by the "/scripts/:token/install-*.sh" scripts.
+func (h *Handler) installScriptHandle(w http.ResponseWriter, r *http.Request, params httprouter.Params) (any, error) {
+	// This is a hack because the router is not allowing us to register "/scripts/install.sh", so we use
+	// the parameter ":token" to match the script name.
+	// Currently, only "install.sh" is supported.
+	if params.ByName("token") != "install.sh" {
+		return nil, trace.NotFound(`Route not found, query "/scripts/install.sh" for the install-only script, or "/scripts/:token/install-node.sh" for the install + join script.`)
+	}
+
+	// TODO(hugoShaka): cache function
+	opts, err := h.installScriptOptions(r.Context())
+	if err != nil {
+		return nil, trace.Wrap(err, "Failed to build install script options")
+	}
+
+	script, err := scripts.GetInstallScript(r.Context(), opts)
+	if err != nil {
+		h.logger.WarnContext(r.Context(), "Failed to get install script", "error", err)
+		return nil, trace.Wrap(err, "getting script")
+	}
+
+	w.WriteHeader(http.StatusOK)
+	if _, err := fmt.Fprintln(w, script); err != nil {
+		h.logger.WarnContext(r.Context(), "Failed to write install script", "error", err)
+	}
+
+	return nil, nil
+}
+
+// installScriptOptions computes the agent installation options based on the proxy configuration and the cluster status.
+// This includes:
+// - the type of automatic updates
+// - the desired version
+// - the proxy address (used for updates).
+// - the Teleport artifact name and CDN
+func (h *Handler) installScriptOptions(ctx context.Context) (scripts.InstallScriptOptions, error) {
+	const group, uuid = "", ""
+
+	version, err := h.autoUpdateAgentVersion(ctx, group, uuid)
+	if err != nil {
+		h.logger.WarnContext(ctx, "Failed to update agent version", "error", err)
+		version = teleport.Version
+	}
+
+	// if there's a rollout, we do new autoupdates
+	_, rolloutErr := h.cfg.AccessPoint.GetAutoUpdateAgentRollout(ctx)
+	if rolloutErr != nil && !trace.IsNotFound(rolloutErr) {
```

Comment:
nit: Unimplemented presumably should also mean that there's no rollout configured in the cluster, right? At least one that we can currently understand.